pkg/gardenlet/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -327,9 +327,17 @@ func (r *reconciler) delete(ctx context.Context, gardenClient client.Client, con
 		conditionInstalled = gardencorev1beta1helper.UpdatedCondition(conditionInstalled, gardencorev1beta1.ConditionFalse, "DeletionFailed", fmt.Sprintf("Deletion of ManagedResource secret %q failed: %+v", controllerInstallation.Name, err))
 	}
 
-	if err := seedClient.Client().Delete(ctx, getNamespaceForControllerInstallation(controllerInstallation)); client.IgnoreNotFound(err) != nil {
+	namespace := getNamespaceForControllerInstallation(controllerInstallation)
+	err = seedClient.Client().Delete(ctx, namespace)
+	if err == nil || apierrors.IsConflict(err) {
```

Comment:
Isn't it better to simply report the conflict as an error, instead of treating it as success?

Reply:
I am not sure what you mean. For K8s < 1.17 deleting a namespace that is already in Terminating returns 409 Conflict. You can see that this is changed with https://github.com/kubernetes/kubernetes/pull/84122 and starting 1.17 404 Not Found is being returned. You can see that we tolerate 409 Conflict also when we clean up the Shoot namespaces - ref https://github.com/gardener/gardener/blob/7145aa25f9523945ec7bfbd257ec9101176ade4c/pkg/operation/botanist/cleanup.go#L159-L160

Reply:
OK, thanks for explaining!